app/src/main/java/woowacourse/shopping/data/product/remote/ProductRemoteImpl.kt

```diff
@@ -0,0 +1,297 @@
+            "name" : "올래 미니 크루아상",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2020/06/[9300000002848]_20200626143224628.jpg",
+            "price" : 5000
+        },
+        {
+            "id" : 24,
+            "name" : "주상절리 파이",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2021/04/[9300000002489]_20210421134243043.jpg",
+            "price" : 5000
+        },
+        {
+            "id" : 25,
+            "name" : "크림치즈 브리오슈 보스톡",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2021/03/[9300000002931]_20210325161934333.jpg",
+            "price" : 5000
+        },
+        {
+            "id" : 26,
+            "name" : "23 서머 그린 머그 355 ml",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2023/04/[9300000004357]_20230428094403109.jpg",
+            "price" : 15000
+        },
+        {
+            "id" : 27,
+            "name" : "바리스타춘식 캐릭터 머그",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2023/01/[9300000004332]_20230130132448085.jpg",
+            "price" : 30000
+        },
+        {
+            "id" : 28,
+            "name" : "리저브 골드 테일 머그",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2021/08/[11123299]_20210804102144453.jpg",
+            "price" : 25000
+        },
+        {
+            "id" : 29,
+            "name" : "스타벅스 브런치 세트",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2022/08/[9300000004214]_20220822132741879.jpg",
+            "price" : 60000
+        },
+        {
+            "id" : 30,
+            "name" : "제주 한라봉 머그 237ml",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2023/03/[9300000004323]_20230307143135598.jpg",
+            "price" : 40000
+        },
+        {
+            "id" : 31,
+            "name" : "베이스볼 머그",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2023/05/[9300000004471]_20230510163213366.jpg",
+            "price" : 34000
+        },
+        {
+            "id" : 32,
+            "name" : "23 서머 기린 머그",
+            "imageUrl" : "https://image.istarbucks.co.kr/upload/store/skuimg/2023/04/[11141873]_20230428125930267.jpg",
+            "price" : 45000
+        }
+    ]
+    """
+        .trimIndent()
+
+    init {
+        initMockData()
+    }
+
+    private fun initMockData() {
+        Thread {
+            if (::mockWebServer.isInitialized) return@Thread
+            mockWebServer = MockWebServer()
+            mockWebServer.start()
+            mockWebServer.url("/")
+            mockWebServer.dispatcher = makeProductMockDispatcher()
+        }.start()
+    }
+
+    override fun getProductById(id: Int): ProductEntity {
+        val products = getAllProducts()
+
+        return products.find { it.id == id }
+            ?: throw java.util.NoSuchElementException("해당 상품은 존재하지 않습니다")
```

Comment:
```suggestion
            ?: throw NoSuchElementException("해당 상품은 존재하지 않습니다")
```
자바가 아닌 코틀린의 Exception을 활용해볼까요?